File: sdks/php/src/Auth/Requests/ResetPasswordDto.php
<?php

namespace NovaBilling\Auth\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class ResetPasswordDto extends JsonSerializableType
{
    /**
     * @var string $token Password reset token
     */
    #[JsonProperty('token')]
    public string $token;

    /**
     * @var string $newPassword New password (min 8 characters)
     */
    #[JsonProperty('newPassword')]
    public string $newPassword;

    /**
     * @param array{
     *   token: string,
     *   newPassword: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->token = $values['token'];
        $this->newPassword = $values['newPassword'];
    }
}

File: sdks/csharp/src/NovaBilling/Auth/Requests/ResetPasswordDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ResetPasswordDto
{
    /// <summary>
    /// Password reset token
    /// </summary>
    [JsonPropertyName("token")]
    public required string Token { get; set; }

    /// <summary>
    /// New password (min 8 characters)
    /// </summary>
    [JsonPropertyName("newPassword")]
    public required string NewPassword { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
